hoomd/mesh.py

```diff
@@ -82,52 +82,31 @@ def _attach_hook(self):
             self._simulation.state._cpp_sys_def, len(self._param_dict["types"]))
 
         self._cpp_obj.setTypes(list(self._param_dict['types']))
-        if self._param_dict._dict["triangulation"] is not None:
-            self._set_triangulation(self._param_dict._dict["triangulation"])
 
         if hoomd.version.mpi_enabled:
             pdata = self._simulation.state._cpp_sys_def.getParticleData()
             decomposition = pdata.getDomainDecomposition()
             if decomposition is not None:
                 # create the c++ Communicator
+                self._cpp_obj.setTriangulation(
```

Comment:
Oh was this the problem with the communicator?

Reply:
Yes, that was the issue!